drivers/seesaw_soil/seesaw_soil.c

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2020 Viktor Gal
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+/**
+ * @ingroup     drivers_seesaw_soil
+ * @{
+ *
+ * @file
+ * @brief       Driver for the Adafruit Seesaw Soil Moisture and Temperature Sensor.
+ *
+ * @author      Viktor Gal <viktor.gal@maeth.com>
+ *
+ * @}
+ */
+
+#include <string.h>
+
+#include "assert.h"
+#include "byteorder.h"
+#include "xtimer.h"
+#include "periph/i2c.h"
+#include "seesaw_soil.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+static int _seesaw_read_regs(const seesaw_soil_t *dev, uint16_t reg_addr, void* data, size_t len) {
+    int r;
+
+    reg_addr = htons(reg_addr);
+    /* send the 16bit register address we want to read with STOP bit */
+    if ((r = i2c_write_bytes(dev->params.i2c, dev->params.addr,
+                             &reg_addr, 2, 0)) < 0) {
+        return r;
+    }
+
+    /* wait for the answer */
+    xtimer_usleep(5000);
```

Comment:
Is that from the data sheet?

Reply:
there's really no docs about this. this is based on the official SDK from adafruit, but there has to be a sleep, otherwise it doesn't work...